src/main/java/com/google/api/codegen/transformer/OutputTransformer.java

```diff
@@ -0,0 +1,103 @@
+import com.google.api.codegen.config.TypeModel;
+import com.google.api.codegen.viewmodel.OutputView;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import java.util.Collections;
+import java.util.List;
+
+class OutputTransformer {
+  private static final String RESPONSE_PLACEHOLDER = "$resp";
+
+  static List<OutputSpec> defaultOutputSpecs(MethodModel method) {
+    if (method.isOutputTypeEmpty()) {
+      return Collections.emptyList();
+    }
+    return Collections.singletonList(
+        OutputSpec.newBuilder().addPrint("%s").addPrint(RESPONSE_PLACEHOLDER).build());
+  }
+
+  static OutputView toView(OutputSpec config, MethodContext context) {
+    Runnable once =
+        new Runnable() {
+          boolean ran;
+
+          @Override
+          public void run() {
+            if (ran) {
+              throw new IllegalArgumentException("only one field of OutputSpec may be set");
+            }
+            ran = true;
+          }
+        };
+
+    OutputView.Builder view = OutputView.newBuilder();
+    if (config.hasLoop()) {
+      once.run();
+      throw new UnsupportedOperationException("loop not implemented yet");
+    }
+    if (config.getPrintCount() > 0) {
+      once.run();
+      view.print(toView(config.getPrintList(), context)).kind(OutputView.Kind.PRINT);
+    }
+    return view.build();
+  }
+
+  private static OutputView.PrintView toView(List<String> config, MethodContext context) {
+    Preconditions.checkArgument(!config.isEmpty(), "print spec cannot be empty");
+    return OutputView.PrintView.newBuilder()
+        .printSpec(context.getNamer().getPrintSpec(config.get(0)))
+        .printArgs(
+            config
+                .subList(1, config.size())
+                .stream()
+                .map(a -> accessor(a, context))
+                .collect(ImmutableList.toImmutableList()))
+        .build();
+  }
+
+  private static OutputView.PrintArgView accessor(String config, MethodContext context) {
+    String[] configElems = config.split("\\.");
+    Preconditions.checkArgument(configElems.length != 0, "field string cannot be empty");
+
+    OutputView.PrintArgView.Builder view = OutputView.PrintArgView.newBuilder();
+    TypeModel type;
+    if (configElems[0].equals(RESPONSE_PLACEHOLDER)) {
+      view.variable(context.getNamer().getSampleResponseVarName());
+      type = context.getMethodModel().getOutputType();
+    } else {
+      throw new UnsupportedOperationException("local variable not implemented yet");
+    }
+
+    ImmutableList.Builder<String> accessors = ImmutableList.builder();
+    for (int i = 1; i < configElems.length; i++) {
+      String fieldName = configElems[i];
+      FieldModel field =
+          Preconditions.checkNotNull(
+              type.getField(fieldName), "type %s does not have field %s", type, fieldName);
+      Preconditions.checkArgument(
+          !field.isRepeated() && !field.isMap(), "%s.%s is not scalar", type, fieldName);
```

Comment:
The first substitution (`type`) is giving me `{[Protobuf FieldModel: {repeated string google.example.library.v1.FindRelatedBooksResponse.names}, Protobuf FieldModel: {string google.example.library.v1.FindRelatedBooksResponse.next_page_token}]}` instead of simply `FindRelatedBooksResponse`. Anyway to fix this? (This will affect the the other precondition check, since this one will go away as per the other comment on this line.)